Moq.AutoMock.Tests/DescribeAsDisposable.cs

```diff
@@ -0,0 +1,96 @@
+﻿using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Linq;
+
+namespace Moq.AutoMock.Tests
+{
+    [TestClass]
+    public class DescribeAsDisposable
+    {
+        [TestMethod]
+        public void It_disposes_all_registered_disposable_instances()
+        {
+            TestDisposable test = new();
+            AutoMocker mocker = new();
+            mocker.Use(test);
+            IDisposable disposable = mocker.AsDisposable();
+
+            disposable.Dispose();
+
+            Assert.IsTrue(test.IsDisposed);
+        }
+
+        [TestMethod]
+        public void It_disposes_using_custom_disposable_instance()
+        {
+            TestDisposable test = new();
+            AutoMocker mocker = new();
+            CustomDisposer disposer = new(mocker);
+            mocker.Use(test);
+            mocker.Use<IAutoMockerDisposable>(disposer);
+            IDisposable disposable = mocker.AsDisposable(); 
+
+            disposable.Dispose();
+
+            Assert.IsTrue(disposer is CustomDisposer);
+            Assert.IsTrue(test.IsDisposed);
+        }
+
+        private sealed class CustomDisposer : IAutoMockerDisposable
```

Comment:
I'm guessing this is the answer to my other question about `IAutoMockerDisposable` being public, but I'm not sure why we'd enable this versus just letting the built-in disposer do its thing.

Reply:
The idea was that is someone wanted to have different logic for disposing of objects (perhaps disposing in a particular order) this would allow them to do that. But perhaps I should not implement features until we have a use case for them?